framework/src/markers/ReporterPointMarker.C

```diff
@@ -0,0 +1,71 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ReporterPointMarker.h"
+
+registerMooseObject("MooseApp", ReporterPointMarker);
+
+InputParameters
+ReporterPointMarker::validParams()
+{
+  InputParameters params = Marker::validParams();
+  params.addClassDescription("Marks the region inside or empty if it contains a reporter defined "
+                             "point for refinement or coarsening.");
+  params.addRequiredParam<ReporterName>("x_coord_name", "reporter x-coordinate name");
+  params.addRequiredParam<ReporterName>("y_coord_name", "reporter y-coordinate name");
+  params.addRequiredParam<ReporterName>("z_coord_name", "reporter z-coordinate name");
+  MooseEnum marker_states = Marker::markerStates();
+  params.addRequiredParam<MooseEnum>(
+      "inside", marker_states, "How to mark elements containing a point");
+  params.addRequiredParam<MooseEnum>(
+      "empty", marker_states, "How to mark elements not containing a point");
+  return params;
+}
+
+ReporterPointMarker::ReporterPointMarker(const InputParameters & parameters)
+  : Marker(parameters),
+    ReporterInterface(this),
+    _inside(parameters.get<MooseEnum>("inside").getEnum<MarkerValue>()),
+    _empty(parameters.get<MooseEnum>("empty").getEnum<MarkerValue>()),
+    _x_coord(getReporterValue<std::vector<Real>>("x_coord_name", REPORTER_MODE_REPLICATED)),
+    _y_coord(getReporterValue<std::vector<Real>>("y_coord_name", REPORTER_MODE_REPLICATED)),
+    _z_coord(getReporterValue<std::vector<Real>>("z_coord_name", REPORTER_MODE_REPLICATED))
```

Comment:
It is a bit awkward to require users to provide all three coordinates even for a 1D and 2D mesh.

Reply:
I'm getting my coordinates from a nodalValueSampler so I always get 3 coordinates.  It is also cleaner to addRequireParams for all three coordinates as opposed to having to check this later on.  
I tried getting this to work for points but there isn't an object that can easily create points